sdk/core/azure-core/azure/core/pipeline/_base.py

```diff
@@ -130,10 +136,11 @@ def __init__(self, transport, policies=None):
         self._transport = transport
 
         for policy in policies or []:
-            if isinstance(policy, SansIOHTTPPolicy):
+            if callable(getattr(policy, "send", None)):
```

Comment:
Guess this deserves a else now that raises that it won't work. Before, we were appended the policy and praying and it would have failed at the first "pipeline.run". Now, it silently drops the object passed as policy (silent drops are dangerous)

Reply:
Sure, that's reasonable. Breaks a few (hacky) tests for Key Vault (fixed by #16784), we'll see what else. Pipeline still needs to silently ignore None in policy lists because Configuration uses that as a default value for policies; I added test coverage of that.